doc/RichTextArea/RichTextArea.md

```diff
@@ -0,0 +1,742 @@
+ │   ├─ RichParagraph          presents the paragraph contents to view
+ │   │   ├─ StyleAttrs         immutable map of style attributes 
+ │   │   ├─ StyledSegment      immutable styled segment
+ │   └─ DataFormatHandler      converter for import/export/clipboard operations
+ │       ├─ StyledInput        input stream of StyledSegments
+ │       ├─ StyledOutput       output stream of StyledSegments
+ │       └─ StyleResolver      converts external CSS styles to attributes
+ │
+ └─ RichTextAreaSkin           default skin
+     └─SideDecorator           right/left paragraph decorator factory
+```
+
+CodeArea extends RichTextArea and brings a few additional classes into the picture:
+
+```
+CodeArea                       control, extends RichTextArea
+ ├─ CodeTextModel              document model, extends BasePlainTextModel, extends StyledTextModel
+ └─ SyntaxDecorator            interface which provides styling for underlying plain text
+```
+
+The code is currently being incubated in the **javafx.incubator.controls module**.
+While this document makes an attempt to give an overview of various parts, please refer to the
+[API Specification](javadoc/javadoc.zip) for more detail.
+
+
+
+
+### Design Principles
+
+- paragraph-oriented model, up to ~2 billion rows
+- virtualized text cell flow
+- supports text styling with an application stylesheet or inline attributes
+- supports multiple views connected to the same model
+- single selection
+- input map allows for easy behavior extension
+
+
+
+### RichTextArea Properties
+
+The new **RichTextArea** control exposes the following properties:
+
+|Property                   |Description                                                                 |Styleable|
+|:--------------------------|:---------------------------------------------------------------------------|:--------|
+|anchorPosition	            |provides the anchor position (read-only)	
+|caretBlinkPeriod	        |determines the caret blink period	
+|caretPosition	            |provides the caret position (read-only)	
+|contentPadding	            |defines the amount of padding in the content area                           |Yes
+|displayCaret	            |indicates whether the caret is displayed	
+|editable	                |indicates whether the editing is enabled	
+|highlightCurrentParagraph	|indicates whether the current paragraph is highlighted
+|leftDecorator	            |specifies the left side paragraph decorator
+|model	                    |document data model	
+|rightDecorator	            |specifies the right side paragraph decorator
+|selectionSegment	        |tracks the selection segment (read-only)
+|useContentHeight	        |determines whether the preferred height is the same as the content height
+|useContentWidth	        |determines whether the preferred width is the same as the content width
+|wrapText	                |indicates whether text should be wrapped                                    |Yes
```

Comment:
`selectedText` in not a good idea in case of large texts, so no.
`promptText` - possibly, though I would rather not add it due to added complexity (we are not dealing with simple text string)
`textFormatter` - no, we are not dealing with a single string.  this is solved by a decorator pattern in the CodeArea, and by the model in general.

`undoable` and `redoable` - makes sense, will add.